vm/src/builtins/dict.rs

```diff
@@ -655,10 +664,46 @@ impl Iterator for DictIter {
     }
 
     fn size_hint(&self) -> (usize, Option<usize>) {
-        let l = self.dict.entries.len_from_entry_index(self.position);
+        let l = self.len();
         (l, Some(l))
     }
 }
+impl ExactSizeIterator for DictIter {
+    fn len(&self) -> usize {
+        self.dict.entries.len_from_entry_index(self.position)
+    }
+}
+
+pub struct DictIterRef<'a> {
```

Comment:
probably `DictRefIter` because  it is iterator of DictRef

```suggestion
pub struct DictRefIter<'a> {
```

Reply:
Changed `DictIter` to `DictIntoIter` (like `std::vec::IntoIter`) and `DictIterRef` to just `DictIter`, to follow existing rust conventions.